docs/artifacts.md

```diff
@@ -155,6 +155,82 @@ spec:
 
 It is recommended to use [purl format](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) for artifacts uri as shown in the example. 
 
+### Output Artifacts in SLSA Provenance
+By default, Tekton Chains will consider all output artifacts as `byProducts` when generating in the [SLSA provenance](https://slsa.dev/spec/v1.0/provenance). In order to treat an artifact as a [subject](https://slsa.dev/spec/v1.0/provenance#schema) of the build, you must set a boolean field `"isBuildArtifact": true` for the output artifact.
+
+e.g.  
+```yaml
+apiVersion: tekton.dev/v1
+kind: TaskRun
+metadata:
+  generateName: step-artifacts-
+spec:
+  taskSpec:
+    description: |
+      A simple task that populates artifacts to TaskRun stepState
+    steps:
+      - name: artifacts-producer
+        image: bash:latest
+        script: |
+          cat > $(artifacts.path) << EOF
+          {
+            "outputs":[
+              {
+                "name":"image",
+                "isBuildArtifact": true,
```

Comment:
I'm worried that "build artifact" isn't specific enough, since users run builds in tasks/pipelines, so any artifact could be referred to as a build artifact.

`buildOutput` is slightly better here to be in line with SLSA terminology of outputs vs byproducts and at least makes a better distinction when talking about different artifact types - e.g. it's weird to say "this artifact isn't a build artifact it's a byproduct" without knowing the context that not all artifacts produced by builds are build artifacts, whereas "this artifact isn't a build output it's a byproduct" feels like it would be less confusing?
There are still similar overlaps with outputs.buildOutput, but this might be the least-worst option? 

Also open to other ideas here! (naming is hard 😅)

Reply:
Thats ok. I'm happy to adopt that.

Reply:
Done!